### ISSUE TITLE
Clicks on Widget with CheckBoxes inside VScrollPanel have incorrect y positions.

### ISSUE THREAD

**Issue by ericmdai** [wjakob/nanogui#98]
When a Widget with CheckBoxes is wrapped in a VScrollPanel, a click within the VScrollPanel has an offset y position and maps to a different CheckBox than a user may intend to click on.

```cpp
#include <nanogui/nanogui.h>

#include <iostream>

using namespace nanogui;
using std::to_string;
using std::cout;

int main(int /* argc */, char ** /* argv */) {
  nanogui::init();

  Screen *screen = new Screen(Vector2i(215, 300), "Issue Demo");
  Window *window = new Window(screen, "VScrollPanel");
  window->setLayout(new GroupLayout());

  VScrollPanel *scroll_panel = new VScrollPanel(window);
  scroll_panel->setFixedSize(Vector2i(125,200));
  Widget *checkboxes = new Widget(scroll_panel);
  checkboxes->setLayout(new BoxLayout(Orientation::Vertical, Alignment::Minimum, 0, 0));
  for(int i = 0; i < 100; i++) {
      CheckBox* checkbox = new CheckBox(checkboxes, "CheckBox " + to_string(i));
      checkbox->setChecked(true);
      checkbox->setCallback([i](bool) { cout << "Clicked CheckBox " << to_string(i) << "\n"; });
  }

  screen->setVisible(true);
  screen->performLayout();
  window->center();

  nanogui::mainloop();

  nanogui::shutdown();
  return 0;
}


```

**Comment by willxinc** [comment]
From what I've seen. you need to have another (empty?) widget inside the window and have the vscrollpanel inside that widget in order for the offsets to be correct

Ex:
```
Widget *outer = new Widget(window);
outer->setFixedSize(Eigen::Vector2i(125, 200));

VScrollPanel *scroll_panel = new VScrollPanel(outer);
scroll_panel->setFixedHeight(200);
```
Instead of:
```
VScrollPanel *scroll_panel = new VScrollPanel(window);
scroll_panel->setFixedSize(Vector2i(125,200));
```

If this is not intended behavior then editing [vscrollpanel.cpp#L74](https://github.com/wjakob/nanogui/blob/master/src/vscrollpanel.cpp#L74) 

from 
```
return mChildren[0]->mouseButtonEvent(p + Vector2i(0, shift), button, down, modifiers);
```
to
```
return mChildren[0]->mouseButtonEvent(p + Vector2i(0, shift - mPos.y()), button, down, modifiers);
```
Also "fixes" this.

**Comment by wjakob** [comment]
I've committed a fix in 227af5b7df90efe846e88ec4c414e72cf90e76ce. Can you verify if this addresses your issue?

**Comment by ericmdai** [comment]
@willxinc Good catch, looks like that works!

@wjakob Looks like it's working now. Thanks for pushing the fix!
